cpp/tests/binaryop/binop-null-test.cpp

```diff
@@ -17,189 +17,122 @@
+  auto rhs = cudf::test::fixed_width_column_wrapper<int32_t>({0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+                                                             cudf::test::iterators::no_nulls());
 
-  using ADD = cudf::library::operation::Add<TypeOut, TypeLhs, TypeRhs>;
+  auto out = cudf::binary_operation(
+    lhs, rhs, cudf::binary_operator::ADD, cudf::data_type(cudf::type_id::INT32));
 
-  auto lhs = make_random_wrapped_column<TypeLhs>(100, mask_state::ALL_NULL);
-  auto rhs = make_random_wrapped_column<TypeRhs>(100, mask_state::ALL_VALID);
-
-  auto out =
-    cudf::binary_operation(lhs, rhs, cudf::binary_operator::ADD, data_type(type_to_id<TypeOut>()));
-
-  ASSERT_BINOP<TypeOut, TypeLhs, TypeRhs>(*out, lhs, rhs, ADD());
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(out->view(), lhs);
 }
 
 TEST_F(BinaryOperationNullTest, Vector_Null_Vector_NonNullable)
 {
-  using TypeOut = int32_t;
-  using TypeLhs = int32_t;
-  using TypeRhs = int32_t;
-
-  using ADD = cudf::library::operation::Add<TypeOut, TypeLhs, TypeRhs>;
+  auto lhs = cudf::test::fixed_width_column_wrapper<int32_t>({0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+                                                             cudf::test::iterators::all_nulls());
+  auto rhs = cudf::test::fixed_width_column_wrapper<int32_t>({0, 1, 2, 3, 4, 5, 6, 7, 8, 9});
 
-  auto lhs = make_random_wrapped_column<TypeLhs>(100, mask_state::ALL_NULL);
-  auto rhs = make_random_wrapped_column<TypeRhs>(100, mask_state::UNALLOCATED);
+  auto out = cudf::binary_operation(
+    lhs, rhs, cudf::binary_operator::ADD, cudf::data_type(cudf::type_id::INT32));
 
-  auto out =
-    cudf::binary_operation(lhs, rhs, cudf::binary_operator::ADD, data_type(type_to_id<TypeOut>()));
-
-  ASSERT_BINOP<TypeOut, TypeLhs, TypeRhs>(*out, lhs, rhs, ADD());
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(out->view(), lhs);
 }
 
 TEST_F(BinaryOperationNullTest, Vector_Valid_Vector_NonNullable)
 {
-  using TypeOut = int32_t;
-  using TypeLhs = int32_t;
-  using TypeRhs = int32_t;
-
-  using ADD = cudf::library::operation::Add<TypeOut, TypeLhs, TypeRhs>;
-
-  auto lhs = make_random_wrapped_column<TypeLhs>(100, mask_state::ALL_VALID);
-  auto rhs = make_random_wrapped_column<TypeRhs>(100, mask_state::UNALLOCATED);
+  auto lhs = cudf::test::fixed_width_column_wrapper<int32_t>({9, 8, 7, 6, 5, 4, 3, 2, 1, 0},
+                                                             cudf::test::iterators::no_nulls());
+  auto rhs = cudf::test::fixed_width_column_wrapper<int32_t>({0, 1, 2, 3, 4, 5, 6, 7, 8, 9});
 
-  auto out =
-    cudf::binary_operation(lhs, rhs, cudf::binary_operator::ADD, data_type(type_to_id<TypeOut>()));
+  auto out = cudf::binary_operation(
+    lhs, rhs, cudf::binary_operator::ADD, cudf::data_type(cudf::type_id::INT32));
 
-  ASSERT_BINOP<TypeOut, TypeLhs, TypeRhs>(*out, lhs, rhs, ADD());
+  auto expected = cudf::test::fixed_width_column_wrapper<int32_t>(
+    {9, 9, 9, 9, 9, 9, 9, 9, 9, 9}, cudf::test::iterators::no_nulls());
```

Comment:
I hope `binary_operator::ADD` is not implemented as `{return 9;}` :P

Side question - how is `fixed_width_column_wrapper` created with `no_nulls()` different from one created without the null mask iterator parameter?

Reply:
The `no_nulls()` creates the null-mask with all valid bits.  Without the iterator, you get a null null-mask -- no null-mask at all.